langkit/templates/pkg_implementation_body_ada.mako

```diff
@@ -170,6 +170,23 @@ package body ${ada_lib_name}.Implementation is
    --  If E is known, return its unique Id from State. Otherwise, assign it a
    --  new unique Id and return it.
 
+   pragma Warnings (Off, "referenced");
+   function Image (Self : Symbol_Type) return ${T.String.name};
+   --  Transform a Symbol into an internal String
+
+   function Image (Self : Symbol_Type) return ${T.String.name} is
+      T : Text_Type := Symbols.Image (Self);
+      Result : constant ${T.String.name} :=
+         ${T.String.constructor_name} (T'Length);
+   begin
+      for I in T'Range loop
+         Result.Items (I - T'First + Result.Items'First) :=
+            ${T.String.element_type.to_internal_expr('T (I)', 'Context')};
```

Comment:
I’m surprised: why is this loop needed? Couldn’t we just have `Result.Items := T;`?